src/include/duckdb/common/types/string_type.hpp

```diff
@@ -108,7 +110,7 @@ struct string_t {
 		} else {
 			// copy the data into the prefix
 #ifndef DUCKDB_DEBUG_NO_INLINE
-			auto dataptr = (char *)GetDataUnsafe();
+			auto dataptr = (char *)GetData();
```

Comment:
```suggestion
			auto dataptr = GetData();
```

GetData should be a const char* already, so all good.
There are another couple of C-style casts are hard to avoid here (although they can probably become const_cast or even better getting the GetDataWritable(), so it become explicit what's happening).